docs/specs/priority-queue.md

```diff
@@ -1,71 +1,73 @@
-# Priority queue proposal
+﻿# Priority queue proposal
 
 ## User data
 
 Before anything is proposed, let's start with what can be prioritized...
 
 ### Input types
 
-We want to be able to support all of the following configurations:
+There are two possible scenarios in which to use a priority queue:
 
-1. User data *is separate* from the priority (two physical instances).
-2. User data *contains* the priority (as one or more properties).
-3. User data *is* the priority (implements `IComparable<T>`).
-4. Rare case: priority is obtainable via some other logic (resides in an object
-different from the user data).
+1. Each item of user data is *assigned* a *separate* priority (priority is a separate object from the elements in the collection.) We will sometimes call this the *extrinsic priority* scenario.
+2. User data either *is itself a priority*, or has some other *natural* priority which can be derived from the element. (It might have a priority property, a it might be a *computed* priority using some function mapping elements to priorities.) We will sometimes call this the *intrinsic priority* scenario.
```

Comment:
> (It might have a priority property, a it might be a *computed* priority using some function mapping elements to priorities.)

This needs to be reworded since "...property, a it might be a ..." doesn't make sense.